src/Middleware/ThrottleMiddleware.php

```diff
@@ -186,6 +186,7 @@ protected function _rateLimit(ThrottleInfo $throttle): RateLimitInfo
         $event = $this->dispatchEvent(self::EVENT_BEFORE_CACHE_SET, [
             'rateLimit' => $rateLimit,
             'ttl' => $ttl,
+            'throttleInfo' => $throttle,
```

Comment:
Can you please add/update a test which asserts that the callback recieves instance of Throttle info as 3rd argument.

Reply:
Provided, test is little bit more extensive, and I've added `clone($throttleInfo)` to prevent side-effects of this event, as that could yield unexpected process flow